### DISCUSSION TITLE
How to start raspap inside the docker container if it is not running? (Running as an addon in homeassistant)

### DISCUSSION THREAD

**Discussion by raphael303** [RaspAP/raspap-docker#40] [General]
Hi 
I have been, unsuccessfully, trying to get RaspAP to run as an addon in HomeAssistant. Since Addons in homeassistant run docker containers and I simply want raspap to run, without any communication with homeassistant, it seemed a straight forward thing to do. I will use a USBmodem and USBWlanInterface for RaspAP

I succeeded to a part, but as to my testing RaspAP was not running within the container. From the command line I found raspap.service, looked at it and within found servicestart.sh . I tried to run raspAP with that, but that relies on systemd which again does not work in the container.

I wonder, how would I start RaspAp in the container, without servicestart.sh?

In case anybody tries to do that, I share here my Homeassistant addonfiles.
Dockerfile:
```
ARG BUILD_FROM
FROM ghcr.io/raspap/raspap-docker:latest

# Copy data for add-on
COPY run.sh /
RUN chmod a+x /run.sh

CMD ["/run.sh"]
```

config.yaml
```
name: "RaspAP"
description: "RaspAP running in a Docker container"
version: "1.0.0"
slug: "raspap"
init: false
arch:
  - armhf
  - armv7
  - aarch64
host_network: true
privileged: true
devices:
  - /dev/ttyUSB0
  - /dev/ttyUSB1
  - /sys/fs/cgroup:/sys/fs/cgroup:rw
map:
  - config:rw
ports:
  80/tcp: 80
```
  
  run.sh
```
  #!/bin/sh

# Run RaspAP
tail -f /dev/null
```

**Comment by billz** [comment]
I believe @NL-TCH has some experience integrating RaspAP-Docker with HomeAssistant. He might be able to advise you here.

**Reply by NL-TCH** [reply]
hey all,
Yeah indeed systemd does not work in docker (for now) without the cgroups fix. The cgroup option is not available in HA.
I however do not know how to start raspap without the service file.
